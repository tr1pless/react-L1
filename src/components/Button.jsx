import React, { Component } from 'react';
import '../App.css'

export const Button = (props) => {
    return <button className='btn' onClick={props.click}>Lets Try</ button>
}