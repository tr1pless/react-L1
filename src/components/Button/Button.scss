.button {
  margin: -140px auto !important;
}
