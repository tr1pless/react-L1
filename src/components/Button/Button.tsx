import React, { FC } from 'react';
import './Button.scss'
import ButtonUI from '@mui/material/Button';

interface ButtonProps {
onButtonClick: () => void;
}

export const Button: FC<ButtonProps> = ({onButtonClick}) => (
        <ButtonUI 
        variant="contained"
        type='submit' 
        onClick={onButtonClick}
        > click 
        </ButtonUI>
    )
