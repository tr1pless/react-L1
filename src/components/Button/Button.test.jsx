import React from 'react'
import { Button } from './Button'
import {render, screen, fireEvent, waitFor} from '@testing-library/react'
import '@testing-library/jest-dom'
import  userEvent from '@testing-library/user-event'

describe('button', () => {

    it('render component', () => {
        render(<Button />)
    })

    it('render with snapshot', () => {
        const {asFragment} = render(<Button />)
        expect(asFragment()).toMatchSnapshot()
    })

    it('render component with text', () => {
        render(<Button />);
        expect(screen.getByText(/click/)).toBeInTheDocument()
    })
    
    it('render multiply components', () => {
        render(<>
        <Button />
        <Button />
        </>)

        expect(screen.queryAllByRole('button').length).toBe(2)
    });

    it('button is disabled', () => {
        render(<Button disabled/>);

        expect(screen.getByText('click')).toBeDisabled()
    })
// 
    // it('button backgroundColor is red', () => {
        // render (<Button />)
// 
        // expect(screen.getByText('click')).toHaveStyle({backgroundColor: 'red'})
    // })

    // it('button click with userEvent', () => {
        // const mockHandler = jest.fn()

        // render (<Button  onButtonClick={mockHandler}/>)

        // userEvent.click(screen.getByText('click'))

        // expect(mockHandler).toBeCalledTime(0)
    // })

    it('button async click', async () => {
        const mockHandler = jest.fn()
        render(<Button onButtonClick={() => setTimeout(mockHandler, 500)}/>)

        userEvent.click(screen.getByText('click'))

        await waitFor(() => expect(mockHandler).toHaveBeenCalledTime(1))
    })
})