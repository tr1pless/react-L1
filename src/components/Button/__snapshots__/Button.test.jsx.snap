// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button render with snapshot 1`] = `
<DocumentFragment>
  <button
    style="background-color: red;"
    type="submit"
  >
    click
  </button>
</DocumentFragment>
`;
